src/WatcherDelegatingHandler.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace k8s
+{
+    internal class WatcherDelegatingHandler : DelegatingHandler
+    {
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            var originResponse = await base.SendAsync(request, cancellationToken);
+
+            if (originResponse.IsSuccessStatusCode)
+            {
+                if ($"{request.RequestUri.Query}".Contains("watch=true"))

```

Comment:
We should probably do better parsing, so that `?baywatch=true-ish` won't trigger this handling.

Reply:
Thanks, 
this must be something I am missing, cloud you please referrer any full doc about watch in k8s.

Reply:
I'm just saying that a user can attach random query strings (possibly accidentally, possibly on purpose) and this code will trigger a watch, even if a watch isn't actually being requested.

We should parse the query string into a map, and then check `map['watch'] == 'true'`

Reply:
I introduce a dependency `Microsoft.AspNetCore.WebUtilities`  to parse the query. I am not sure you might consider it too heavy.

another approach is 

```
if ($"&{request.RequestUri.Query}&".Contains("&watch=true&"))
|| ($"&{request.RequestUri.Query}&".Contains("&?watch=true&"))
```

seems too ugly